src/samVisualize/commands/visualizeSamTemplate/samVisualization.ts

```diff
@@ -0,0 +1,227 @@
+    private readonly extensionContext: vscode.ExtensionContext
+
+    // To let the extension know a visualization has been closed
+    private readonly onVisualizationDisposeEmitter = new vscode.EventEmitter<void>()
+
+    constructor(textDocument: vscode.TextDocument, extensionContext: vscode.ExtensionContext) {
+        this.extensionContext = extensionContext
+        this.textDocument = textDocument
+        this.textDocumentUri = textDocument.uri
+        this.webviewPanel = this.createWebiewPanel(textDocument)
+    }
+
+    /**
+     * Reveals the webviewPanel for this visualization
+     */
+    public revealPanel(): void {
+        this.webviewPanel?.reveal()
+    }
+
+    private createWebiewPanel(textDocument: vscode.TextDocument): vscode.WebviewPanel | undefined {
+        const panel = vscode.window.createWebviewPanel(
+            'samVisualization',
+            `${basename(textDocument.fileName)} (Rendering)`,
+            {
+                preserveFocus: true,
+                viewColumn: vscode.ViewColumn.Beside,
+            },
+            {
+                enableScripts: true,
+                retainContextWhenHidden: true,
+            }
+        )
+        // During tests, a mock context is used, so the absolute paths will not resolve
+        try {
+            const graphObject = generateGraphFromYaml(textDocument.getText())
+
+            // Only generate a ResourceLineMap if the input has generated a valid GraphObject.
+            if (graphObject) {
+                this.resourceLineMap = generateResourceLineMap(textDocument.getText())
+            }
+
+            const graphObjectString = JSON.stringify(graphObject)
+
+            const primaryResourceListString = readFileSync(
+                this.extensionContext.asAbsolutePath(
+                    join('resources', 'light', 'samVisualize', 'primaryResources.json')
+                )
+            ).toString()
+
+            const iconsDir = this.extensionContext.asAbsolutePath(join('resources', 'light', 'samVisualize', 'icons'))
+
+            const iconsDataString = JSON.stringify(generateIconsMap(iconsDir, panel.webview))
+
+            const webviewStylesheetUri = panel.webview.asWebviewUri(
+                vscode.Uri.file(this.extensionContext.asAbsolutePath(join('media', 'css', 'samVisualize.css')))
+            )
+
+            const webviewBodyScriptUri = panel.webview.asWebviewUri(
+                // Change those file names
+                vscode.Uri.file(
+                    this.extensionContext.asAbsolutePath(join('samVisualizeRenderJS', 'samVisualizeRenderBundle.js'))
+                )
+            )
+            panel.webview.html = this.getWebviewContent(
+                graphObjectString,
+                primaryResourceListString,
+                iconsDataString,
+                webviewStylesheetUri,
+                webviewBodyScriptUri,
+                panel.webview.cspSource
+            )
+        } catch {}
```

Comment:
What could potentially be thrown and why isn't it a concern? Please add a comment in the catch block.

Reply:
I'll move the comment down to the catch block. During testing, a mock extensionContext is used, so the resource paths will not resolve.